extensions/ql-vscode/src/skeleton-query.ts

```diff
@@ -0,0 +1,218 @@
+import { join } from "path";
+import { CancellationToken, Uri, workspace } from "vscode";
+import { CodeQLCliServer } from "./cli";
+import { OutputChannelLogger } from "./common";
+import { Credentials } from "./common/authentication";
+import { QueryLanguage } from "./common/query-language";
+import { askForLanguage, isFolderAlreadyInWorkspace } from "./helpers";
+import { getErrorMessage } from "./pure/helpers-pure";
+import { QlPackGenerator } from "./qlpack-generator";
+import { DatabaseItem, DatabaseManager } from "./local-databases";
+import * as databaseFetcher from "./databaseFetcher";
+import { ProgressCallback } from "./progress";
+
+type QueryLanguagesToDatabaseMap = {
+  [id: string]: string;
+};
+
+export class SkeletonQueryWizard {
+  private language: string | undefined;
+  private folderName: string | undefined;
```

Comment:
I'm happy to convert this, but I'm not sure what the advantage would be in this case? 

As far as I understand, `getter` methods are a way to access private members of a class. `folderName` is already private and I don't really want to make this accessible from outside.

Also adding this code doesn't seem like it would improve anything:

```
  private _folderName: string | undefined;
  ...
  public get folderName() {
    return this._folderName;
  }
```

Happy to be told otherwise though, just not sure this is the right use case (as far as my understanding goes of this).

Reply:
That is one way of using getters, but getters can also return dynamic values. I was thinking more along the lines of:

```ts
private get folderName() {
  return `codeql-custom-queries-${this.language}`;
}
```

This removes some instance state from the class, which would make this class easier to understand in my opinion.

Reply:
Ah that makes more sense! Cool, I'll add it. Thanks!